src/br/com/zup/nossacasacodigo/author/Author.java

```diff
@@ -36,7 +34,55 @@ public String getDescription() {
 		return this.description;
 	}
 	
-	public Date getCreatedAt() {
+	public LocalDateTime getCreatedAt() {
 		return this.createdAt;
 	}
+	
+	//Valida��es
+	private void emailValidator(String email) {
+		isEmpty(email);
+
+		boolean isEmailValid = false; 
+		if (email != null && email.length() > 0) {
+			//Determinamos uma express�o padr�o que determina um e-mail v�lido atrav�s da String expression
+			String expression = "^[\\w\\.-]+@([\\w\\-]+\\.)+[A-Z]{2,4}$";
+			//Definimos nossa express�o como um padr�o e a tornamos sens�vel a letras maiusculas
+			Pattern pattern = Pattern.compile(expression, Pattern.CASE_INSENSITIVE);
+			//verificamos se o e-mail inserido obedece ao padr�o determinado
+			Matcher matcher = pattern.matcher(email);
+			if (matcher.matches()) {
+				isEmailValid = true;
+			}
+		}
+		
+		if(isEmailValid == false) {
+			throw new IllegalArgumentException();
+		}
+	}
+
+	private void nameValidator(String name) {
+		isEmpty(name);
+	}
+
+	private void descriptionValidator(String description) {
+		isEmpty(description);
+
+		if(description.length() > 400) {
+			throw new IllegalArgumentException();
+		}
+	}
+
+	private void isEmpty(String string) {
+		if (string == null) {
+			throw new RuntimeException("Erro de Preenchimento de String");
```

Comment:
**Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.**

Aqui não é questão de não usar hehe, mas a RuntimeException é uma classe de nome genérico. Nesse caso você já deixou a mensagem que dá uma dica do problema. O tipo da exception entra como complemento. Lembra que quando um software pode parar de funcionar numa situação meio tensa... milhares de pessoas usando, tempo passando, gente reclamando etc... Quanto mais fácil for achar o problema, melhor.